sw/device/tests/rv_core_ibex_mem_test.c

```diff
@@ -158,24 +180,37 @@ static void setup_flash(void) {
 bool test_main(void) {
   setup_uart();
 
-  LOG_INFO("Testing Load from ROM Location.");
-  uint32_t load = *kRomTestLoc;
-  CHECK(load == kRomTestLocContent,
-        "The content of the ROM test location was 0x%08x and not the expected "
-        "value.",
-        load);
-
-  use_icache(false);
-  LOG_INFO("Running an instruction from ROM with icache disabled.");
-  rom_test_gadget();
-
-  use_icache(true);
-  LOG_INFO("Running an instruction from ROM with icache enabled.");
-  rom_test_gadget();
+  // ROM access is blocked in the silicon owner stage.
+  if (kBootStage != kBootStageOwner) {
+    LOG_INFO("Testing Load from ROM Location.");
+
+    // For the execution test we a specific `c.jr x1` (i.e. function return)
+    // instruction. Since the address can vary between ROM builds, we scan a
+    // small region to find it.
+    volatile uint16_t *test_loc;
+    for (test_loc = (uint16_t *)kRomTestLocStart;
+         test_loc < (uint16_t *)kRomTestLocEnd; test_loc++) {
+      if (*test_loc == kRomTestLocContent) {
+        break;
+      }
+    }
```

Comment:
Nit. A tiny tweak to make it easier to read in my option. But feel free to ignore. 

```suggestion
    volatile uint16_t *test_loc = (uint16_t *)kRomTestLocStart;
    while(test_loc < (uint16_t *)kRomTestLocEnd && *test_loc != kRomTestLocContent) {
       test_loc++;
    }
```

Reply:
I personally prefer the current structure as I think it better reflects the "loop through all words and search" intent